#!/usr/bin/env python
PACKAGE = "laser_touch_screen"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("origin_frame_id",                           str_t,     0, "origin_frame_id",                             "origin")
gen.add("screen_origin_frame_id",                    str_t,     0, "screen_origin_frame_id",                      "screen_origin")
gen.add("top_left_laser_frame_id",                   str_t,     0, "top_left_laser_frame_id",                     "top_left_laser")
gen.add("lower_right_laser_frame_id",                str_t,     0, "lower_right_laser_frame_id",                  "lower_right_laser")
gen.add("endpoint_dist_threshold",                double_t,     0, "endpoint_dist_threshold",                     0.02,     -10.0,  10.0)
gen.add("with_threshold_l",                       double_t,     0, "with_threshold_l",                            0.03,     -10.0,  10.0)
gen.add("with_threshold_h",                       double_t,     0, "with_threshold_h",                            0.15,     -10.0,  10.0)

gen.add("top_left_frame_id",                          str_t,    0, "top_left_frame_id",                           "top_left_scan")
gen.add("top_left_scan_topic_name",                   str_t,    0, "top_left_scan_topic_name",                    "/top_left_scan")
gen.add("top_left_module_namespace",                  str_t,    0, "top_left_module_namespace",                   "top_left")
gen.add("top_left_pub_end_point_en",                 bool_t,    0, "top_left_pub_end_point_en",                   True)
gen.add("top_left_scan_origin_x",                  double_t,    0, "top_left_scan_origin_x",                      0.00,     -10.0,  10.0)
gen.add("top_left_scan_origin_y",                  double_t,    0, "top_left_scan_origin_y",                      0.00,     -10.0,  10.0)
gen.add("top_left_scan_origin_z",                  double_t,    0, "top_left_scan_origin_z",                      0.00,     -10.0,  10.0)
gen.add("top_left_scan_yaw",                       double_t,    0, "top_left_scan_yaw",                           0.00,     -3.14,  3.14)
gen.add("top_left_scan_pitch",                     double_t,    0, "top_left_scan_pitch",                         0.00,     -3.14,  3.14)
gen.add("top_left_scan_roll",                      double_t,    0, "top_left_scan_roll",                          0.00,     -3.14,  3.14)

gen.add("lower_right_frame_id",                       str_t,    0, "lower_right_frame_id",                        "lower_right_scan")
gen.add("lower_right_scan_topic_name",                str_t,    0, "lower_right_scan_topic_name",                 "/lower_right_scan")
gen.add("lower_right_module_namespace",               str_t,    0, "lower_right_module_namespace",                "lower_right")
gen.add("lower_right_pub_end_point_en",              bool_t,    0, "lower_right_pub_end_point_en",                True)
gen.add("lower_right_scan_origin_x",               double_t,    0, "lower_right_scan_origin_x",                   0.00,     -10.0,  10.0)
gen.add("lower_right_scan_origin_y",               double_t,    0, "lower_right_scan_origin_y",                   0.00,     -10.0,  10.0)
gen.add("lower_right_scan_origin_z",               double_t,    0, "lower_right_scan_origin_z",                   0.00,     -10.0,  10.0)
gen.add("lower_right_scan_yaw",                    double_t,    0, "lower_right_scan_yaw",                        0.00,     -3.14,  3.14)
gen.add("lower_right_scan_pitch",                  double_t,    0, "lower_right_scan_pitch",                      0.00,     -3.14,  3.14)
gen.add("lower_right_scan_roll",                   double_t,    0, "lower_right_scan_roll",                       0.00,     -3.14,  3.14)

exit(gen.generate(PACKAGE, "laser_touch_screen", "laser_touch_screen_"))
